std/container/array.d

```diff
@@ -364,6 +364,9 @@ private struct RangeT(A)
     assert(S.s_nDestroyed == S.s_nConstructed);
 }
 
+//Fix https://issues.dlang.org/show_bug.cgi?id=23140
+private alias Unshared(T) = T;
+private alias Unshared(T: shared U, U) = U;
```

Comment:
I think that unshared should either error out or return the same type that was passed when used with a type that is shared at the aggregate definition level (because such an object was defined to work only as shared). Using unshared on such objects just opens the door to a lot of problems. In my opinion, this should be fixed at the language level and make it impossible to cast away `shared` (or any other qualifiers) if an object is defined as such.

Reply:
Will do.  Turns out there is no need for Unshared at all, just a cast(void*) in the GC.free calls (which doesn't break the code, it still frees the memory).

Reply:
I can't remove Unshared completely, because otherwise opEquals will refuse to compile, and the workaround (comparing each element individually) I came up with also errors out for the same reason as the existence of this regression.